### PR TITLE
Modified scatter method to correctly recognize RGB/A values

### PR DESCRIPTION
**Pull request by vecuenca**
This is a fix for #5377.

After being converted to a ndarray, the scatter method does not check if c_array is a 2D array (i.e. if we are passing RGB or RGB/A values.

### CODE REVIEW

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -3861,7 +3861,7 @@ def scatter(self, x, y, s=20, c=None, marker='o', cmap=None, norm=None,
         # favor of mapping, not rgb or rgba.
         try:
             c_array = np.asanyarray(c, dtype=float)
-            if c_array.size == x.size:
+            if c_array.size == x.size and c_array.shape[0] != 1:
```
Comment (tacaswell) [inline lib/matplotlib/axes/_axes.py]: Is there anyway we can get a numpy-scaler in here as a valid input?  If so probably want to be a bit more paranoid and add `and carray.ndims`

Comment (tacaswell) [inline lib/matplotlib/axes/_axes.py]: and would the check on `c_array.ndim == 1` be a better check?

**Comment by vincent-cuenca** [comment]
Yes, I think you are right. I hadn't noticed the existence of ndim previously.

**Comment by tacaswell** [comment]
Can you please rebase onto current master (or at least current v2.x)?

Merging upstream branches into feature branches causes problems, please do not do it.

**Comment by mdboom** [comment]
Closing this.  There doesn't seem to be a good way to resolve this ambiguity, and it's already been discussed and notated in the comment in the code where this change is.

**Pull request closed without merging**